File: src/main/java/nl/chimpgamer/networkmanager/extensions/discordbot/commands/mc/NetworkManagerBotCommand.java
package nl.chimpgamer.networkmanager.extensions.discordbot.commands.mc;

import nl.chimpgamer.networkmanager.extensions.discordbot.DiscordBot;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.MCMessage;
import nl.chimpgamer.networkmanager.api.commands.NMBungeeCommand;
import nl.chimpgamer.networkmanager.api.sender.Sender;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class NetworkManagerBotCommand extends NMBungeeCommand {
    private final DiscordBot discordBot;

    public NetworkManagerBotCommand(DiscordBot discordBot, String cmd) {
        super(discordBot.getNetworkManager(), cmd, Collections.singletonList("networkmanager.admin"), "nmbot");
        this.discordBot = discordBot;
    }

    @Override
    public void onExecute(Sender sender, String[] args) {
        if (args.length == 2) {
            if (args[0].equalsIgnoreCase("reload")) {
                switch (args[1].toLowerCase()) {
                    case "config":
                    case "settings":
                        getDiscordBot().getSettings().reload();
                        sender.sendMessage(MCMessage.RELOAD_CONFIG.getMessage());
                        break;
                    case "messages":
                        getDiscordBot().getMessages().reload();
                        sender.sendMessage(MCMessage.RELOAD_MESSAGES.getMessage());
                        break;
                    case "jda":
                        boolean success = getDiscordBot().getDiscordManager().restartJDA();
                        if (success) {
                            sender.sendMessage(MCMessage.RELOAD_JDA_SUCCESS.getMessage());
                        } else {
                            sender.sendMessage(MCMessage.RELOAD_JDA_FAILED.getMessage());
                        }
                        break;
                }
            }
        }
    }

    @Override
    public List<String> onTabComplete(Sender sender, String[] args) {
        if (args.length == 1 && "reload".startsWith(args[0].toLowerCase())) {
            return Collections.singletonList("reload");
        }
        if (args.length == 2) {
            return Arrays.asList("config", "settings", "messages", "jda");
        }
        return Collections.emptyList();
    }

    private DiscordBot getDiscordBot() {
        return discordBot;
    }
}

File: src/main/kotlin/nl/chimpgamer/networkmanager/extensions/discordbot/commands/mc/NetworkManagerBotCommand.kt
package nl.chimpgamer.networkmanager.extensions.discordbot.commands.mc

import nl.chimpgamer.networkmanager.api.models.sender.Sender
import nl.chimpgamer.networkmanager.api.utils.commands.NMBungeeCommand
import nl.chimpgamer.networkmanager.extensions.discordbot.DiscordBot
import nl.chimpgamer.networkmanager.extensions.discordbot.configurations.MCMessage

class NetworkManagerBotCommand(private val discordBot: DiscordBot, cmd: String) : NMBungeeCommand(discordBot.networkManager, cmd, listOf("networkmanager.admin"), listOf("nmbot")) {
    override fun onExecute(sender: Sender, args: Array<String>) {
        if (args.size == 2) {
            if (args[0].equals("reload", ignoreCase = true)) {
                when (args[1].toLowerCase()) {
                    "config", "settings" -> {
                        discordBot.settings.reload()
                        sender.sendMessage(discordBot.messages.getString(MCMessage.RELOAD_CONFIG))
                    }
                    "messages" -> {
                        discordBot.messages.reload()
                        sender.sendMessage(discordBot.messages.getString(MCMessage.RELOAD_MESSAGES))
                    }
                    "jda" -> {
                        val success = discordBot.discordManager.restartJDA()
                        if (success) {
                            sender.sendMessage(discordBot.messages.getString(MCMessage.RELOAD_JDA_SUCCESS))
                        } else {
                            sender.sendMessage(discordBot.messages.getString(MCMessage.RELOAD_JDA_FAILED))
                        }
                    }
                }
            }
        }
    }

    override fun onTabComplete(sender: Sender, args: Array<String>): List<String> {
        return if (args.size == 1 && "reload".startsWith(args[0], ignoreCase = true)) {
            listOf("reload")
        } else if (args.size == 2) {
            listOf("config", "settings", "messages", "jda")
        } else emptyList()
    }
}